instrumentation/servlet/src/main/java/io/opentelemetry/auto/instrumentation/servlet/dispatcher/RequestDispatcherInstrumentation.java

```diff
@@ -95,17 +92,15 @@ public static SpanWithScope start(
     @Advice.OnMethodExit(onThrowable = Throwable.class, suppress = Throwable.class)
     public static void stop(
         @Advice.Enter final SpanWithScope spanWithScope,
-        @Advice.Local("_requestSpan") final Object requestSpan,
+        @Advice.Local("_originalServletSpan") final Object originalServletSpan,
         @Advice.Argument(0) final ServletRequest request,
         @Advice.Thrown final Throwable throwable) {
       if (spanWithScope == null) {
         return;
       }
 
-      if (requestSpan != null) {
-        // now add it back...
-        request.setAttribute(SPAN_ATTRIBUTE, requestSpan);
-      }
+      // restore the original servlet span
+      request.setAttribute(SPAN_ATTRIBUTE, originalServletSpan);
```

Comment:
Why remove the null check?

Reply:
I pushed a clarifying comment just now, does this makes sense?

```
// since spanWithScope is non-null here, originalServletSpan must have been set with the prior
// servlet span (as opposed to remaining unset)
```